pkg/webhook/mutation/pod/otlp/mutator.go

```diff
@@ -0,0 +1,234 @@
+package otlp
+
+import (
+	"fmt"
+	"github.com/Dynatrace/dynatrace-operator/pkg/clients/dynatrace"
+	"github.com/Dynatrace/dynatrace-operator/pkg/consts"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/otlp"
+	dtwebhook "github.com/Dynatrace/dynatrace-operator/pkg/webhook/mutation/pod/mutator"
+	corev1 "k8s.io/api/core/v1"
+	"slices"
+)
+
+const (
+	envVarOtlpTraceEndpoint    = "OTEL_EXPORTER_OTLP_TRACES_ENDPOINT"
+	envVarOtlpLogsEndpoint     = "OTEL_EXPORTER_OTLP_LOGS_ENDPOINT"
+	envVarOtlpMetricsEndpoint  = "OTEL_EXPORTER_OTLP_METRICS_ENDPOINT"
+	envVarOtlpExporterEndpoint = "OTLP_EXPORTER_OTLP_ENDPOINT"
+
+	envVarOtlpTraceProtocol    = "OTEL_EXPORTER_OTLP_TRACES_PROTOCOL"
+	envVarOtlpLogsProtocol     = "OTEL_EXPORTER_OTLP_LOGS_PROTOCOL"
+	envVarOtlpMetricsProtocol  = "OTEL_EXPORTER_OTLP_METRICS_PROTOCOL"
+	envVarOtlpExporterProtocol = "OTLP_EXPORTER_OTLP_PROTOCOL"
+
+	envVarOtlpTraceHeaders    = "OTEL_EXPORTER_OTLP_TRACES_HEADERS"
+	envVarOtlpLogsHeaders     = "OTEL_EXPORTER_OTLP_LOGS_HEADERS"
+	envVarOtlpMetricsHeaders  = "OTEL_EXPORTER_OTLP_METRICS_HEADERS"
+	envVarOtlpExporterHeaders = "OTLP_EXPORTER_OTLP_HEADERS"
+
+	envVarOtlpTraceCertificate    = "OTEL_EXPORTER_OTLP_TRACES_CERTIFICATE"
+	envVarOtlpLogsCertificate     = "OTEL_EXPORTER_OTLP_LOGS_CERTIFICATE"
+	envVarOtlpMetricsCertificate  = "OTEL_EXPORTER_OTLP_METRICS_CERTIFICATE"
+	envVarOtlpExporterCertificate = "OTLP_EXPORTER_OTLP_CERTIFICATE"
+
+	// TODO make sure naming does not collide with anything
+	activeGateTrustedCertVolumeName    = "activegate-server-certs"
+	activeGateTrustedCertSecretKeyPath = "activegate.pem"
+	activeGateTrustedCertMountPath     = "activegate-server-certs"
+	envActiveGateTrustedCert           = activeGateTrustedCertMountPath + "/" + activeGateTrustedCertSecretKeyPath
+
+	envVarDtApiToken = "DT_API_TOKEN"
+
+	otlpAPIPath = "/v2/otlp"
+)
+
+type OTLPMutator struct{}
+
+func NewMutator() dtwebhook.Mutator {
+	return &OTLPMutator{}
+}
+
+func (OTLPMutator) IsEnabled(request *dtwebhook.BaseRequest) bool {
+	//TODO implement when structure for OTLP enrichment config in Dynakube has been defined.
+	return true
+}
+
+func (OTLPMutator) IsInjected(request *dtwebhook.BaseRequest) bool {
```

Comment:
this is more of a "legacy" part of the interface

it is mainly used for differentiating between: 
- Mutate -> changes need to be made to the init-container (general setup, add volumes, etc) + mutate the user container
- Reinvoke -> Only (new) container needs some attention, the init-container is fine
   - History: It used to be that reinvoke still touched the init-container, but not the volumes and such, this has been changed. 

For otelp specifically, you only care if the envs are on all the containers you want to "inject", no init-container to worry about